java/libs/src/main/java/org/physical_web/collection/PhysicalWebCollection.java

```diff
@@ -324,7 +324,7 @@ public PwPair getTopRankedPwPairByGroupId(String groupId, Comparator<PwPair> com
         // Create the group if it doesn't exist
         UrlGroup urlGroup = groupMap.get(groupId);
         if (urlGroup == null) {
-          urlGroup = new UrlGroup(groupId);
+          urlGroup = new UrlGroup(groupId, comparator);
```

Comment:
I don't think the group itself should know about the comparator.  Just pass it as a param to getTopPair